pkg/scheduler/framework/parallelize/parallelism.go

```diff
@@ -51,15 +51,27 @@ func chunkSizeFor(n, parallelism int) int {
 	return s
 }
 
+// workersForChunkSize returns number of workers (goroutines)
+// that will be created in workqueue.ParallelizeUntil
+// for given parallelism, pieces and chunkSize values.
+func workersForChunkSize(parallelism, pieces, chunkSize int) int {
+	chunks := (pieces + chunkSize - 1) / chunkSize
+	if chunks < parallelism {
+		return chunks
+	}
+	return parallelism
+}
+
 // Until is a wrapper around workqueue.ParallelizeUntil to use in scheduling algorithms.
 // A given operation will be a label that is recorded in the goroutine metric.
 func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece workqueue.DoWorkPieceFunc, operation string) {
 	goroutinesMetric := metrics.Goroutines.WithLabelValues(operation)
 	withMetrics := func(piece int) {
-		goroutinesMetric.Inc()
 		doWorkPiece(piece)
 		goroutinesMetric.Dec()
 	}
+	chunkSize := chunkSizeFor(pieces, p.parallelism)
+	goroutinesMetric.Add(float64(workersForChunkSize(p.parallelism, pieces, chunkSize)))
```

Comment:
Should we have a comment here to mention why we write it this way is to ensure performance?